src/services/services.ts

```diff
@@ -2912,6 +2919,63 @@ module ts {
 
                 return filteredMembers;
             }
+
+            function isExternalModuleName(token: Node): boolean {
+                if (token && token.parent && token.kind === SyntaxKind.StringLiteral && isInStringOrRegularExpressionOrTemplateLiteral(token)) {
+                    return token.parent.kind === SyntaxKind.ImportDeclaration ||
+                        token.parent.kind === SyntaxKind.ExportDeclaration ||
+                        token.parent.kind === SyntaxKind.ExternalModuleReference;
+                }
+                return false;
+            }
+
+            function getExternalModulecompletions(location: Node): CompletionInfo {

```

Comment:
Leave a comment about what kinds of completions we support. For instance, we don't hit the disk to look up available files (and note this in the comment), so be explicit, since I find that modules are the most arcane part of the compiler and language.